libs/payments/currency/src/lib/currency.manager.ts

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Injectable } from '@nestjs/common';
+import {
+  CURRENCIES_TO_COUNTRIES,
+  SUPPORTED_PAYPAL_CURRENCIES,
+  VALID_COUNTRY_CODES,
+  VALID_CURRENCY_CODES,
+} from './currency.constants';
+import {
+  CurrencyCodeInvalidError,
+  CountryCodeInvalidError,
+  CurrencyCountryMismatchError,
+} from './currency.error';
+
+@Injectable()
+export class CurrencyManager {
+  constructor() {}
+
+  /**
+   * Verify that provided source country and plan currency are compatible with
+   * valid currencies and countries as listed in constants
+   *
+   * @param currency Currency of customer
+   * @param country Country of customer
+   * @returns True if currency is compatible with country, else throws error
+   */
+  assertCurrencyCompatibleWithCountry(
+    currency: string | null | undefined,
+    country: string | null | undefined
+  ): void {
+    if (!currency) throw new CurrencyCodeInvalidError(currency);
+    if (!country) throw new CountryCodeInvalidError(country);
+
+    const currencyToCountryMap: Record<string, string[]> = {
+      ...CURRENCIES_TO_COUNTRIES,
+    };
```

Comment:
[non-blocking] this object copy is a little deceptive. Since it's not deep copied, someone might think that this map is safe to mutate, but the array of countries is still the global array reference from the constants file - if modified, it would modify it globally.

Reply:
I was getting type errors, hence why I opted to do it this way. But I can totally update. Let's discuss this afternoon. 👍

Reply:
Updated as per our Slack conversation! Thank you for the suggested change!